### PR TITLE
Expose fifoFilesFallbackPath as a config property.

### PR DESCRIPTION
**Pull request by cmelchior**
Closes https://github.com/realm/realm-js/issues/2062

Example of a working configuration:
```
        const dirs = RNFetchBlob.fs.dirs;
        const realmPath = dirs.DownloadDir + '/test.realm';
        const fallbackDir = dirs.CacheDir;
        Realm.open({
            fifoFilesFallbackPath: fallbackDir,
            path: realmPath,
            readOnly: false,
            schema: [DogSchema]
        }).then(realm => {
            // ...
        });
```

## ☑️ ToDos
* [x] Changelog
* [x] Wait for https://github.com/realm/realm-object-store/pull/765 to be merged and redirect
* [x] Verify manually that this fixes External Storage on Android.
* [x] Check that path is a directory and path ends with `/` (Done in ObjectStore)

### CODE REVIEW

File: docs/realm.js [hunk 1]
```diff
@@ -371,6 +371,12 @@ class Realm {
  *   will be skipped if another process is accessing it.
  * @property {string} [path={@link Realm.defaultPath}] - The path to the file where the
  *   Realm database should be stored.
+ * @property {string} [fifoFilesFallbackPath] - Opening a Realm creates a number of FIFO files in order to
```
Comment (bmunkholm) [inline docs/realm.js]: Perhaps we don't need to expose the internal details in particular in the naming, but suffice to have a generically named admin path?
perhaps `writableFallbackPath`? Or just `writablePath`?

File: docs/realm.js [hunk 2]
```diff
@@ -371,6 +371,12 @@ class Realm {
...
+ * coordinate access to the Realm across threads and processes. If the Realm file is stored in a location
+ * that does not allow the creation of FIFO files (e.g. FAT32 filesystems), then the Realm cannot be opened.
+ * In that case Realm needs a different location to store these files and this property defines that location.
+ * The FIFO files are very lightweight and the main Realm file will still be stored in the location defined
+ * by the `path` property.
```
Comment (bmunkholm) [inline docs/realm.js]: Will it be ignored if the defaultPath is writable?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@ x.x.x Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
 * The parser now supports readable timestamps with a `T` separator in addition to the originally supported `@` separator. For example: `startDate > 1981-11-01T23:59:59:1`. ([realm/realm-core#3198](https://github.com/realm/realm-core/issues/3198))
+* It is now possible to store Realms on Android external storage with React Native by using `Realm.Configuration.fifoFallbackPath()`. ([#2062](https://github.com/realm/realm-js/issues/2062)).
```
Comment (nirinchev) [inline CHANGELOG.md]: `Realm.Configuration.fifoFallbackPath()` makes it look like a method that needs to be called.

File: docs/realm.js (hunk 1 above)
Comment (cmelchior) [reply docs/realm.js]: Normally I would agree, but in this case, it is directly related to the files being FIFO's. Also, this option should be rarely used, so making it as descriptive as possible is actually desirable IMO.

**Comment by cmelchior** [comment]
Ready for review

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -116,3 +116,21 @@ To finish adding your new function, you will have to add your function a few pla
 * Documentation is added in `docs/realm.js`
 * Add an entry to `CHANGELOG.md` if applicable (Breaking changes/Enhancements/Bug fixes)
 
+### Updating Object Store
+
+Object Store is the cross platform abstraction shared between all language bindings supported by Realm.
```
Comment (kneth) [inline CONTRIBUTING.md]:
```suggestion
Object Store is the cross platform abstraction shared between all SDKs supported by Realm.
```

Comment (kneth) [inline CONTRIBUTING.md]:
```suggestion
[Object Store](https://github.com/realm/realm-object-store) is the cross platform abstraction shared between all language bindings supported by Realm.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -116,3 +116,21 @@ To finish adding your new function, you will have to add your function a few pla
...
+It is found [here](https://github.com/realm/realm-object-store).
```
Comment (kneth) [inline CONTRIBUTING.md]:
```suggestion
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -116,3 +116,21 @@ To finish adding your new function, you will have to add your function a few pla
...
+
+It is included in Realm JS as a Git submodule under `/src/object-store`.
```
Comment (kneth) [inline CONTRIBUTING.md]:
```suggestion
It is included in Realm JS as a Git submodule under `src/object-store`.
```

**Pull request merged**